### PR TITLE
Use new description generation

### PR DESCRIPTION
**Pull request by znarf**
- Remove hardcoded descriptions and use the generator in Sequelize hooks
- Fix a few typos in generator
- Fix directly related tests (snapshots and looking at description string)
- Fix indirectly broken tests (somehow the returned order is changing for Transactions)

### CODE REVIEW

File: server/models/Transaction.js [hunk 1]
```diff
@@ -257,6 +258,16 @@ function defineModel() {
       },
 
       hooks: {
+        beforeCreate: async transaction => {
+          if (!transaction.description) {
+            transaction.description = await generateDescription(transaction);
```
Comment (Betree) [inline server/models/Transaction.js:263]: This centralized approach will make it more difficult to wrap the creation of new transactions in SQL transactions (what was started in https://github.com/opencollective/opencollective-api/pull/5981). We can eventually take the one from `beforeCreate` arguments and pass it down to `generateDescription` but that will make things more complex (what if you're using loaders?).

In this regard, the existing approach of setting the description where the transaction gets created makes more sense as these parts of the code already have the data needed to generate the descriptions, they don't need to preload any additional information.

We can still add some helpers when they're used in multiple places, for example:

```
models.Transaction.create({
  isRefund: true,
  description: generateRefundTransactionDescription(originalTransaction),
})
```

**Review by Betree — COMMENTED** [review]
Do we want to migrate past data?

**Comment by znarf** [comment]
> Do we want to migrate past data?

For now, I don't think so. Main reason is that the generator is software based and we would need to iterate 1 by 1 on all transactions in the table.

File: server/models/Transaction.js (hunk 1 above)
Comment (znarf) [reply server/models/Transaction.js:263]: Ok, let's take some time to think about this.

**Pull request closed without merging**